src/endpoints/http/collections/export/index.js

```diff
@@ -0,0 +1,143 @@
+import archive from './lib/archive';
+import UnsupportedMediaError from '../../../../Errors/UnsupportedMediaError';
+
+const GB = 1073741824; // this will be configurable through fh-config
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {db} db - The db connection.
+ * @param {String} collectionName - The name of the collection requested to get the size of.
+ * @returns Promise
+ */
+function getCollectionSizeOne(db, collectionName) {
+  return new Promise((resolve, reject) => {
+    db.collection(collectionName).stats((err, stats) => {
+      if (err) {
+        return reject(err.message === 'ns not found' ? new Error(`${collectionName} collection does not exist`) : err);
+      }
+
+      resolve(stats.size);
+    });
+  });
+}
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {db} db - The db connection.
+ * @param {String} collectionNames - The String array of collections requested to export.
+ * @returns Promise
+ */
+function getCollectionsSize(db, collectionNames) {
+  return Promise.all(collectionNames.map(name => getCollectionSizeOne(db, name)));
+}
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {Number} sizes - Number array containing each collections size.
+ * @returns {Number} - sum of all the collection sizes.
+ */
+function getTotalSize(sizes) {
+  return sizes.reduce((acc, size) => acc + size, 0);
+}
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {String} collections - String array of all collection names.
+ * @param {db} db - db connection.
+ * @param {String} format - The requested format to export the collections.
+ * @param {Boolean} raw - true for a collection's documents to be returned in bson format, false for json.
+ * @returns {Stream} - An array of streams containing each collection's documents.
+ */
+function getCollectionStreams(collections, db, format, raw) {
+  return collections.map(name => {
+    const collection = db.collection(name);
+    const cursor = collection.find({}, { raw: raw });
+    cursor.filename = `${name}.${format}`;
+    return cursor.stream();
+  });
+}
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {Stream} streams - An array of streams containing each collection's documents.
+ * @param {String} format - The requested format to export the collections.
+ * @returns {Stream} - An array of streams containing each collection's documents parsed in the requested format.
+ */
+function setParsers(streams, format) {
+  return streams.map(collection => {
+    try {
+      const parsedCollections = parsers[format](collection);
+      return parsedCollections;
+    } catch (err) {
+      throw new UnsupportedMediaError();
+    }
+  });
+}
+
+/**
+ * Gets the size of each collection requested to export.
+ *
+ * @param {Stream} zipFile - An array of streams containing each collection's documents parsed in the requested format.
```

Comment:
I think you meant `A zip file stream containing each collection's documents parsed in the requested format`?